### PR TITLE
Add "remove search query" icon/ button to blog search

### PR DESCRIPTION
**Pull request by seann1**
I have added the clear search functionality to the blog search bar. I used a fontawesome icon on the right of the search input which triggers the clear search function when clicked. I added the column search/categories layout at the same time. I hope this isn't a problem. I needed to add the fontawesome css file and webfonts folder to get the icon to align properly to the right of the search input.

### CODE REVIEW

**Review by s-gbz — CHANGES_REQUESTED** [review]
Hey there Sean, sorry for the late-ish reply and thanks for your PR  :100: .
Please try to avoid commits to the Gem- & lock.file, because it messes with the netlify build.
You can do that by only staging specific files like `blog.html` in our case.

Our first follow up task would be to hence remove the changes to `Gemfile.lock`.
This also goes for all new fontawesome files, because they tend to be unnecessary weight to the build since we only use a couple of icons.

You can however add your desired icon via svg.
[Here's](https://github.com/codeklasse/codeklasse.de/blob/bcc605a652b37b94d4328751c8ea3805adc5a19a/_includes/footer.html#L35) an example how you can do that. It will blend in seamlessly, just make sure to give it a width as seen in the example.

I will provide final technical feedback after the requested changes if you don't mind.
As again, let me know if there's anything I can provide you and rock on :rocket: :relaxed:

File: blog.html [hunk 1]
```diff
@@ -156,13 +165,24 @@
...
         }
     }
+
+    function clearSearch() {
+        const searchInput = document.getElementById("searchInput");
+        searchInput.value = "";
+        const elems = document.querySelectorAll(".hidden");
+        [].forEach.call(elems, function(el) {
+            el.classList.remove("hidden");
+        });
```
Comment (s-gbz) [inline blog.html:184]:
```suggestion
        elems.forEach(el => {
            el.classList.remove("hidden");
        });
```

File: blog.html [hunk 2]
```diff
@@ -156,13 +165,24 @@
...
             }
+
         }
     }
+
+    function clearSearch() {
+        const searchInput = document.getElementById("searchInput");
+        searchInput.value = "";
+        const elems = document.querySelectorAll(".hidden");
+        elems.forEach(el => {
```
Comment (s-gbz) [inline blog.html:182]:
```suggestion
        const hiddenPosts = document.querySelectorAll(".hidden");
        hiddenPosts.forEach(el => {
```

**Review by s-gbz — APPROVED** [review]
I improved the naming & used a fat arrow function.
Other than that thanks for helping out @seann1 - the layout works really fine!

**Comment by s-gbz** [comment]
Juuuuuuuuust in case you happened to be bored, there is a very last blog related issue #30
As always, feel free to ignore it and have a wonderful week! :sun_with_face: 
All the best from Germany :beers:

**Comment by seann1** [comment]
Thanks, and thanks for the help!

**Pull request merged**